shell/src/main/java/alluxio/shell/command/LsCommand.java

```diff
@@ -128,20 +135,30 @@ private void ls(AlluxioURI path, boolean recursive, boolean forceLoadMetadata, b
     }
     List<URIStatus> statuses = listStatusSortedByIncreasingCreationTime(path, options);
     for (URIStatus status : statuses) {
-      System.out.print(formatLsString(SecurityUtils.isSecurityEnabled(), status.isFolder(),
-          FormatUtils.formatMode((short) status.getMode(), status.isFolder()), status.getOwner(),
-          status.getGroup(), status.getLength(), status.getCreationTimeMs(),
-          100 == status.getInMemoryPercentage(), status.getPath()));
+      if (pinnedMetadata) {
+        if (status.isPinned()) {
+          System.out.print(formatLsString(SecurityUtils.isSecurityEnabled(), status.isFolder(),
```

Comment:
this block is the same as the `else` block, and appears in other places, could you consider abstracting it out as a private method.